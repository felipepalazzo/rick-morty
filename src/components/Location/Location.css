.text-container {
	font-size: 0.95rem;
	line-height: 1.4;
}

.text-container div + div {
	margin-top: 0.5rem;
}
