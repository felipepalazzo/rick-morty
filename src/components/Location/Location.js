import React from 'react'
import PropTypes from 'prop-types'
import './Location.css'

export default function Location(props) {
	return (
		<section className="text-container">
			<div>
				<strong>Planet:</strong> {props.name}
			</div>
			<div>
				<strong>Dimension:</strong> {props.dimension}
			</div>
			<div>
				<strong>Residents:</strong> {props.residents.length}
			</div>
		</section>
	)
}

Location.defaultProps = {
	residents: []
}
