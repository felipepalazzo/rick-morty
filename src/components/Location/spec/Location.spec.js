import React from 'react'
import { shallow } from 'enzyme'
import Location from '../Location'

describe('Location', () => {
	const props = { name: 'Earth', dimension: 'other' }
	it('renders without crashing', () => {
		const wrapper = shallow(<Location {...props} />)
		expect(wrapper).toMatchSnapshot()
	})
})
