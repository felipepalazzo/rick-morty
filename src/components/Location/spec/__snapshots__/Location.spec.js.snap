// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location renders without crashing 1`] = `ShallowWrapper {}`;
